src/script/engine.zig

```diff
@@ -841,3 +842,57 @@ test "Script execution - OP_SHA256" {
     Sha256.hash(&[_]u8{1}, &expected_hash, .{});
     try std.testing.expectEqualSlices(u8, expected_hash[0..], hash_bytes);
 }
+
+test "ScriptBuilder Smoke test" {
+    const allocator = std.testing.allocator;
+    var sb = try ScriptBuilder.new(allocator);
+    sb.deinit();
+
+    var engine = try (try (try (try sb.pushInt(1)).pushInt(2)).pushOpcode(Opcode.OP_ADD)).build();
```

Comment:
Correct me if I'm wrong but I tried this out and using `?` means the function should return an option type. Currently, it returns either the ScriptBuilder or an error.
Should I modify the function to return an option type? this would mean I need to handle the errors manually in the function itself, and return NULL if an error is encountered

Reply:
you are right. It's a rust syntax not a zig one.
Mb